src/core_plugins/kibana/public/dashboard/dashboard_strings.js

```diff
@@ -0,0 +1,48 @@
+import { DashboardViewMode } from './dashboard_view_mode';
+import _ from 'lodash';
+
+export class DashboardStrings {
+
+  /**
+   * @param list {Array.<string>}
+   * @returns {string} The list of strings concatenated with commas so it can be used in a message.
+   * E.g. ['a', 'b', 'c'] returns 'a, b, and c'.
+   */
+  static createStringList(list) {
+    const listClone = _.clone(list);
+    const isPlural = list.length > 1;
+    const lastEntry = isPlural ? `, and ${list[list.length - 1]}` : '';
+    if (isPlural) listClone.splice(-1, 1);
+
+    return `${listClone.join(', ')}${lastEntry}`;
+  }
+
+  /**
+   * @param changedFilters {Array.<string>} An optional list of filter types that have changed.
+   * @returns {string} A warning message to display to the user that they are going to lose changes.
+   */
+  static getUnsavedChangesWarningMessage(changedFilters) {
+    const changedFilterList = this.createStringList(changedFilters);
+
+    return changedFilterList ?
+      `Are you sure you want to cancel and lose changes, including changes made to your ${changedFilterList}?` :
+      `Are you sure you want to cancel and lose changes?`;
+  }
+
+  /**
+   * @param title {string} the current title of the dashboard
+   * @param viewMode {DashboardViewMode} the current mode. If in editing state, prepends 'Editing ' to the title.
+   * @param isDirty {boolean} if the dashboard is in a dirty state. If in dirty state, adds (unsaved) to the
+   * end of the title.
+   * @returns {string} A title to display to the user based on the above parameters.
+   */
+  static getDashboardTitle(title, viewMode, isDirty) {
+    const isEditMode = viewMode === DashboardViewMode.EDIT;
+    const unsavedSuffix = isEditMode && isDirty
+      ? ' (unsaved)'
+      : '';
+
+    const displayTitle = `${title}${unsavedSuffix}`;
+    return isEditMode ? 'Editing ' + displayTitle : displayTitle;
+  }
+}
```

Comment:
I also appreciate the reasoning! I totally get your point about namespacing, e.g. `Math.round()` and `$.ajax()`. I think I'm just confused about the role of `static` methods in the context of JavaScript, and when they're appropriate vs. the approach Kim suggests. I did some Googling but couldn't find any satisfactory answers. This is such a nitpicky topic that I'm OK with how the code currently is -- I don't think it's worth debating. 😄